lms/djangoapps/courseware/access.py

```diff
@@ -146,13 +158,19 @@ def _can_access_descriptor_with_start_date(user, descriptor, course_key):  # pyl
 
     Arguments:
         user (User): the user whose descriptor access we are checking.
-        descriptor (AType): the descriptor for which we are checking access.
-    where AType is any descriptor that has the attributes .location and
-        .days_early_for_beta
+        descriptor (AType): the descriptor for which we are checking access,
+            where AType is CourseDescriptor, CourseOverview, or any other class that
+            represents a descriptor and has the attributes .location, .id, .start,
+            and .days_early_for_beta.
+
+    Returns:
+        AccessResponse: The result of this access check. Possible results are access granted or a StartDateError.
     """
+    # pylint: disable=translation-of-non-string
     start_dates_disabled = settings.FEATURES['DISABLE_START_DATES']
-    if start_dates_disabled and not is_masquerading_as_student(user, course_key):
-        return True
+    masquerading = is_masquerading_as_student(user, course_key)
+    if start_dates_disabled and not masquerading:
```

Comment:
This is a merge conflict.  You should revert lines 171-172 to the combined line that it was before.  Otherwise, it results in an extra query.  (That is, `is_masquerading_as_student` is called even if `start_dates_disabled`.)